pkg/minikube/cruntime/cri.go

```diff
@@ -108,10 +110,22 @@ func listCRIContainers(cr CommandRunner, root string, o ListContainersOptions) (
 	}
 
 	args = append(args, "list", "-f", "json")
-	rr, err = cr.RunCmd(exec.Command("sudo", args...))
-	if err != nil {
+
+	// avoid "no such file or directory" runc list error by retrying
```

Comment:
I am generally not a big fan of retyring till it works...I rather waiting for a condition to be met,
if runc is gonna include that in their release soon https://github.com/opencontainers/runc/pull/3349 
I rather wait for that

Reply:
@medya in a less likely, but still occasional scenario where the race condition happens, runc currently exits with an error and we don't have a specific condition to wait for, so we retry and the odds to hit race condition again are minor (but we could repeat a couple of times, if that's needed)

i agree it would be better if this would be handled internally by runc, which that pr aims to address, so i [asked](https://github.com/opencontainers/runc/pull/3349#issuecomment-2021659300) author & approver if there are plans to release it (given that it was merged to the main 2 years ago)